src/actions/channels/CHANNEL_CREATE.ts

```diff
@@ -0,0 +1,22 @@
+import { Action } from '../../lib/structures/Action';
+import { Channel } from '../../client/caching/structures/channels/Channel';
+import { isGuildChannel } from '../../util/Util';
+
+import type { ChannelCreateDispatch } from '@klasa/ws';
+
+export default class CoreAction extends Action {
+
+	public check(): Channel | null {
+		return null;
+	}
+
+	public build(data: ChannelCreateDispatch): Channel | null {
+		return Channel.create(this.client, data.d, data.d.guild_id && this.client.guilds.get(data.d.guild_id));
+	}
+
+	public cache(data: Channel): void {
+		if (isGuildChannel(data)) data.guild.channels.set(data.id, data);
+		else this.client.channels.set(data.id, data);
```

Comment:
Cache redundancy is a painful thing to do, it takes too much memory for no benefit.
Besides, all of Discord's events send you the `guild_id` alongside the `channel_id`, so not necessary to continue what Discord's v6 didn't allow (where some guild events didn't send it).

Reply:
If we are doing that, it might be better named Client#dms, as not to be confused with how client#channels is perceived to work. If we feel it's needed, we can add a dummy client#channels that searches dms/guilds.

Reply:
Done in 507900! 🎉